dags/springer/repository.py

```diff
@@ -0,0 +1,54 @@
+import pathlib
+from typing import IO
+
+import boto3
+import yaml
+from botocore.client import Config
+from mypy_boto3_s3.service_resource import _Bucket
+
+from common.repository import IRepository
+
+
+class SpringerRepository(IRepository):
+    ZIPED_DIR: str = "raw/"
+    EXTRACTED_DIR: str = "extracted/"
+    s3: _Bucket = None
+
+    def __init__(self) -> None:
+        super().__init__()
+        config_path : pathlib.Path = pathlib.Path(__file__).parent.parent.resolve()
+        with open(f"{config_path}/config.yaml", "r") as config_file:
+            config = yaml.safe_load(config_file)['publishers']['springer']['repo']
+            self.s3 = boto3.resource(
+                's3',
+                endpoint_url=config['source'],
+                aws_access_key_id=config['username'],
+                aws_secret_access_key=config['password'],
+                config=Config(signature_version='s3v4'),
+                region_name=config['region']
+            ).Bucket(config['bucket'])
```

Comment:
I would abstract also the use of boto configuration

Reply:
Abstracting the boto config will lock us later if we want to switch the repository source. Could it happen later ? If not I'll put the boto config in the interface aswell 👍